File: Puzzles.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    int n,m,a[100],best = 1000;
    cin>>n>>m;
    for(int i=0;i<m;i++)
    {
        cin>>a[i];
    }
    sort(a, a+m);

    for(int i= 1;i<m-n;i++)
    {
        best = min(best, a[i+n-1]- a[i]);
    }
    if(best==1000)
        best = 0;

    cout<<best;

}

File: Puzzles.py
n , m = map(int, input().strip().split())
f = list(map(int, input().strip().split()))

f.sort()

min_diff = float('inf')

for i in range(m-n+1):
    dif = f[i+n-1]-f[i]
    min_diff = min(min_diff,dif)

print(min_diff)